kubelet/datadog_checks/kubelet/kubelet.py

```diff
@@ -252,26 +252,43 @@ def _perform_kubelet_check(self, instance_tags):
 
     def _report_pods_running(self, pods, instance_tags):
         """
-        Reports the number of running pods on this node
-        tagged by service and creator.
+        Reports the number of running pods on this node and the running
+        containers in pods, tagged by service and creator.
 
         :param pods: pod list object
         :param instance_tags: list of tags
         """
-        tag_counter = {}
+        pods_tag_counter = {}
+        containers_tag_counter = {}
         for pod in pods['items']:
+            # Pod reporting
             pod_id = pod.get('metadata', {}).get('uid')
             tags = get_tags('kubernetes_pod://%s' % pod_id, False) or None
             if not tags:
                 continue
             tags += instance_tags
             hash_tags = tuple(sorted(tags))
-            if hash_tags in tag_counter.keys():
-                tag_counter[hash_tags] += 1
+            if hash_tags in pods_tag_counter.keys():
```

Comment:
this can be just `if hash_tags in pods_tag_counter:`

Alternatively, you can make `pods_tag_counter` a `defaultdict(int)` and get rid completely of the `if/else` block since `pods_tag_counter[hash_tags] += 1` would always work